packages/kbn-config-schema/src/types/never_type.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { internals } from '../internals';
+import { Type, TypeOptions } from './type';
+
+export type NeverOptions = TypeOptions<never> & {
+  message?: string;
```

Comment:
__note:__ I don't have a strong opinion on that, I'm a bit skeptical about having `message` for `never` type while we don't have anything like that for any other type. What does make `never` type so special that it should have a custom error message?

Having said that, the ability to override messages on a call-basis is a great thing (e.g. to hide sensitive data from error message if needed), but I'd rather see this as a part of `TypeOptions` that is shared with every type, but it'd be a good candidate for a dedicated PR.

Reply:
will add as requested